mungegithub/mungers/issue-creator.go

```diff
@@ -0,0 +1,396 @@
+	RealConfig() *github.Config
+}
+
+//gihubConfig is a type alias of github.Config that implements the RepoClient interface.
+type githubConfig github.Config
+
+func (config *githubConfig) RealConfig() *github.Config {
+	return (*github.Config)(config)
+}
+
+func (config *githubConfig) getOrg() string {
+	return config.Org
+}
+
+func (config *githubConfig) getProject() string {
+	return config.Project
+}
+
+func (config *githubConfig) getDryRun() bool {
+	return config.DryRun
+}
+
+func (config *githubConfig) GetUser(login string) (*githubapi.User, error) {
+	return ((*github.Config)(config)).GetUser(login)
+}
+
+func (config *githubConfig) GetLabels() ([]*githubapi.Label, error) {
+	return ((*github.Config)(config)).GetLabels()
+}
+
+func (config *githubConfig) ListAllIssues(options *githubapi.IssueListByRepoOptions) ([]*githubapi.Issue, error) {
+	return ((*github.Config)(config)).ListAllIssues(options)
+}
+
+func (config *githubConfig) NewIssue(title, body string, labels, owners []string) (*github.MungeObject, error) {
+	return ((*github.Config)(config)).NewIssue(title, body, labels, owners)
+}
+
+//Issue is an interface implemented by structs that can be synced with github issues via the IssueCreator.
+type Issue interface {
+	//Title yields the initial title text of the github issue.
+	Title() string
+	//Body yields the body text of the github issue and *must* contain the output of ID().
+	//closedIssues is a (potentially empty) slice containing all closed
+	//issues authored by this bot that contain ID() in their body.
+	//if Body returns an empty string no issue is created.
+	Body(closedIssues []*github.MungeObject) string
+	//ID returns a string that uniquely identifies this issue.
+	//This ID must appear in the body of the issue.
+	//DO NOT CHANGE how this ID is formatted or duplicate issues will be created
+	//on github for this issue
+	ID() string
+	//Labels specifies the set of labels to apply to this issue on github.
+	Labels() []string
+	//Owners returns the github usernames to assign the issue to or nil/empty for no assignment.
+	Owners() []string
+	//Priority calculates and returns the priority of this issue
+	//The returned bool indicates if the returned priority is valid and can be used
+	Priority() (int, bool)
+}
+
+//IssueCreator handles syncing identified issues with github issues.
+//This includes finding existing github issues, creating new ones, and ensuring that duplicate
+//github issues are not created.
+type IssueCreator struct {
+	//config is the github client that is used to interact with github.
+	config RepoClient
+	//validLabels is the set of labels that are valid for the current repo (populated from github).
+	validLabels []*githubapi.Label
+	//authorName is the name of the current bot.
+	authorName string
+	//allIssues is a local cache of all issues in the repo authored by the currently authenticated user.
+	//MungeObjects are keyed by issue number
+	allIssues map[int]*github.MungeObject
```

Comment:
I originally did have that map type, but I switched it to MungeObject so that an Issue interface implementation can get more info about the github issues if it needs it.  Specifically I was thinking that it might be useful to retrieve the comments for the issue which is a functionality provided by MungeObject.  Should I switch it back?